### DISCUSSION TITLE
Run code on server startup

### DISCUSSION THREAD

**Discussion by ricardobeat** [sveltejs/kit#3657] [Q&A]
When using `adapter-node`, is there a way to run code on server startup?

I'm trying to add [Datadog integration](https://docs.datadoghq.com/tracing/setup_overview/setup/nodejs/?tab=containers) for monitoring the service, so far the only viable approach seems to be to wrap the build in my own server.js, which I'd like to avoid if possible - the code is straightforward, but requires installing that server's dependencies separately since it's not in the build folder, and this makes deployment much longer.

**Comment by mustofa-id** [comment]
have you tried run it in `hooks` file?
